mtbl/fileset.c

```diff
@@ -290,11 +304,29 @@ fs_reinit_merger(struct mtbl_fileset *f)
 		f->merger = mtbl_merger_init(f->mopt);
 	}
 	assert(f->merger != NULL);
-	while (my_fileset_get(f->shared_fs->my_fs, i++, &fname, (void **) &reader))
-		if ((reader != NULL) && ((f->fname_filter == NULL) ||
-					 f->fname_filter(fname, f->fname_filter_clos))) {
+	while (my_fileset_get(f->shared_fs->my_fs, i++, &fname, (void **) &reader)) {
+		if (reader == NULL) {
+			continue;
+		}
+
+		/*
```

Comment:
I think that the proposed wording tracks with the logic of the `if` statement. That said, the logic is more complex than needed, and:
```
if ( (f->fname_filter == NULL || f->fname_filter(fname, ...)) &&
      (f->reader_filter == NULL || f->reader_filter(reader, ...))) {
```
would suffice. The prose description of this would be "Add the reader's source to the merger unless filtered out by the fname_filter or reader_filter."

Reply:
Chris has better code and a correspondingly simpler and still accurate comment.

Reply:
I changed the if logic and comment. Thanks Chris!